edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/EdgeDeploymentStatusCommand.cs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes.EdgeDeployment
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.Agent.Core;
+    using Microsoft.Azure.Devices.Edge.Agent.Core.ConfigSources;
+    using Microsoft.Azure.Devices.Edge.Agent.Kubernetes;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class EdgeDeploymentStatusCommand : ICommand
+    {
+        readonly Option<EdgeDeploymentStatus> activeStatus;
+
+        public string Id => "EdgeDeploymentStatusCommand";
+
+        public EdgeDeploymentStatusCommand(
+            Option<EdgeDeploymentStatus> activeStatus)
+        {
+            this.activeStatus = activeStatus;
+        }
+
+        public Task ExecuteAsync(CancellationToken token)
+        {
+            // If the EdgeDeployment operator fails, report this to the plan executor
+            // as an exception.
+            this.activeStatus.ForEach(status =>
+                {
+                    if (status.State != EdgeDeploymentStatusType.Success)
+                    {
+                        throw new ConfigOperationFailureException(status.Message);
+                    }
+                });
```

Comment:
This sounds like reducing all failures into one scenario and all what we expect the operator to do it to report the failure. I understand that the ReportDeploymentFailure method only reports the failure without an attempt to recover from it. Will it be helpful that the exception class characterize a failure type to propagate useful information about the failure? For example the Exception can have the following types:

1. TemporaryError; signals that the exception handler can retry the deployment
2. PermanentError; signals that the exception handler must permanently fail the deployment

That's also means that the plan executor should have a mechanism to retry (note: I did not read that part of the code yet).

Reply:
> This sounds like reducing all failures into one scenario and all what we expect the operator to do it to report the failure

That is approximately how the single-device edge handles problems from command execution. Exceptions are caught and this is reduced to success or failure. However, C# has a notion of [fatal errors](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.devices.common.exceptionextensions.isfatal?view=azure-dotnet), and I should probably allow agent to crash if I get a fatal error. 

Overall, though, the idea of temporary errors vs. permanent may be useful - We have generally left it up to the plan runner to determine how often to retry commands, usually by progressively backing off, but usually the failures do land in one of these two categories.